src/spec/template_spec.js

```diff
@@ -0,0 +1,333 @@
+            "bodyType": "rawHtml",
+            "_links": []
+          }
+        ],
+        "pageSize": 200,
+        "itemsCount": 1,
+        "currentPage": 1,
+        "pagesCount": 1,
+        "_links": []
+      });
+    }));
+  }
+
+  function setupEditTemplateSuccessResponse() {
+    browser.addMockModule('descartableModule6', () => angular
+    .module('descartableModule6', ['ngMockE2E'])
+    .run($httpBackend => {
+
+      $httpBackend.whenPUT(/\/accounts\/[\d|-]*\/templates\/[\w|-]*/).respond(200, {
+        "message": "Template successfully updated",
+        "_links": []
+      });
+    
+      $httpBackend.whenPUT(/\/accounts\/[\d|-]*\/templates\/[\w|-]*\/body/).respond(200, {
+        "message": "Template successfully updated",
+        "_links": []
+      });
+    }));
+  }
+
+  function setupEditTemplateErrorResponse() {
+    browser.addMockModule('descartableModule7', () => angular
+    .module('descartableModule7', ['ngMockE2E'])
+    .run($httpBackend => {
+
+      $httpBackend.whenPUT(/\/accounts\/[\d|-]*\/templates\/[\w|-]*/).respond(500, {
+        "message": "Template successfully updated",
+        "_links": []
+      });
+    
+      $httpBackend.whenPUT(/\/accounts\/[\d|-]*\/templates\/[\w|-]*\/body/).respond(500, {
+        "message": "Template successfully updated",
+        "_links": []
+      });
+    }));
+  }
+
+  it('should show the save template button in edit mode', () => {
+    // Arrange
+    beginAuthenticatedSession();
+    browser.get('/#/templates/32e93fa4-c8f4-467a-a2bb-da95f72a27b4');
+    setupTemplateResponse();
+
+    var templatePage = new TemplatePage();
+
+    // Act
+    var saveTemplateButtonIsDisplayed = templatePage.isSaveButtonDisplayed();
+
+    // Assert
+    expect(saveTemplateButtonIsDisplayed).toBeTruthy();
+  });
+
+  it('should not allow to save template with empty body content', () => {
+    // Arrange
+    beginAuthenticatedSession();
+    browser.get('/#/templates/32e93fa4-c8f4-467a-a2bb-da95f72a27b4');
+    setupTemplateResponse();
+
+    var templatePage = new TemplatePage();
+
+    // Act
+    templatePage.cleanTemplateBodySection();
+    templatePage.clickSaveTemplateButton();
+
+    // Assert
+    expect(templatePage.isInvalidTemplateBody()).toBeTruthy();
+  });
+
+  it('should save template in edit mode and redirect to templates list', () => {
+    // Arrange
+    beginAuthenticatedSession();
+    browser.get('/#/templates/32e93fa4-c8f4-467a-a2bb-da95f72a27b4');
+    setupTemplateResponse();
+    setupEditTemplateSuccessResponse();
+    setupTemplateMenuSuccessResponse();
+    var body = "Test content body";
+    var templatePage = new TemplatePage();
+
+    // Act
+    templatePage.fillTemplateBodySection(body);
+    templatePage.clickSaveTemplateButton();
+    
+    // Assert
+    expect(browser.getCurrentUrl()).toEqual(browser.baseUrl + '/#/templates');
+  });
+
+  it('should save template in edit mode and fail', () => {
```

Comment:
What do you think about: _It should show error and do not redirect when save fails_ ?

Reply:
Thanks for the help I take it is more detailed.